### PR TITLE
[BEAM-345] (roll forward) Base Passert on GroupByKey

### PR DESCRIPTION
**Pull request by kennknowles**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by kennknowles** [comment]
R: @amitsela note little fix to spark runner

Other than that, pushing this up to trigger full test matrix. Not necessarily expecting pass.

**Comment by kennknowles** [comment]
R: @dhalperi

I have successfully run the `RunnableOnService` tests on the Dataflow runner. I expect postcommit to be happy with this one.

It turns out the fairly simple NPE in Spark (fixed in the first commit here) was the only thing left.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/testing/PAssert.java [hunk 1]
```diff
@@ -594,31 +718,32 @@ public PDone apply(PBegin input) {
 
       input
           .apply(Create.of(0).withCoder(VarIntCoder.of()))
-          .apply(ParDo.named("RunChecks").withSideInputs(actual)
-              .of(new CheckerDoFn<>(checkerFn, actual)));
+          .apply(
+              ParDo.named("RunChecks")
```
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/testing/PAssert.java]: optional switch to apply syntax

File: sdks/java/core/src/main/java/org/apache/beam/sdk/testing/PAssert.java [hunk 2]
```diff
@@ -594,31 +718,32 @@ public PDone apply(PBegin input) {
...
    * <p>The input is ignored, but is {@link Integer} to be usable on runners that do not support
    * null values.
    */
-  private static class CheckerDoFn<ActualT> extends DoFn<Integer, Void> {
+  private static class SideInputCheckerDoFn<ActualT> extends DoFn<Integer, Void> {
     private final SerializableFunction<ActualT, Void> checkerFn;
     private final Aggregator<Integer, Integer> success =
         createAggregator(SUCCESS_COUNTER, new Sum.SumIntegerFn());
     private final Aggregator<Integer, Integer> failure =
         createAggregator(FAILURE_COUNTER, new Sum.SumIntegerFn());
```
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/testing/PAssert.java]: counters are no longer used?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/testing/PAssert.java [hunk 3]
```diff
@@ -641,87 +763,89 @@ public void processElement(ProcessContext c) {
...
-  public static class TwoSideInputAssert<ActualT, ExpectedT>
-      extends PTransform<PBegin, PDone> implements Serializable {
-
-    private final transient PTransform<PBegin, PCollectionView<ActualT>> createActual;
-    private final transient PTransform<PBegin, PCollectionView<ExpectedT>> createExpected;
-    private final AssertRelation<ActualT, ExpectedT> relation;
+  private static class GroupedValuesCheckerDoFn<ActualT> extends DoFn<ActualT, Void> {
+    private final SerializableFunction<ActualT, Void> checkerFn;
+    private final Aggregator<Integer, Integer> success =
+        createAggregator(SUCCESS_COUNTER, new Sum.SumIntegerFn());
```
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/testing/PAssert.java]: where used?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/testing/PAssert.java (hunk 2 above)
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/testing/PAssert.java]: nm

**Comment by dhalperi** [comment]
LGTM, please self-merge

**Pull request merged**